src/session.ts

```diff
@@ -144,10 +141,23 @@ export class Session extends Client implements ErdstallSession {
 	}
 
 	async leave(): Promise<TransactionGenerator> {
+		let skipped = 0;
+		let cb: ErdstallEventHandler<"phaseshift">;
+		const p = new Promise<void>((accept) => {
+			cb = () => {
+				// One Epoch when the current epoch ends for which we receive the ExitProof.
+				// Two Epochs to guarantee that our ExitProof is part of a sealed epoch.
+				if (skipped <= 2) {
```

Comment:
```suggestion
				if (skipped < 2) {
```
Otherwise you're waiting **3** phase shifts, no?

Reply:
Yes as described in the comment above. First PhaseShift in which we initially receive our `ExitProof` for Epoch `n`. Then two more PhaseShifts for epoch `n+1` and `n+2`. Or am I missing something?

Reply:
relative epoch `n+2` should already be the withdrawal phase, so waiting two phase shifts should suffice.

Reply:
I misunderstood. Using `skipped < 2` works! Thanks for clearing this up!